lmfdb/users/pwdmanager.py

```diff
@@ -146,14 +160,34 @@ def authenticate(self, pwd):
         """
         # from time import time
         # t1 = time()
-        if not 'password' in self._data:
+        if not 'password' in self._data and not 'bcpassword' in self._data:
             logger.warning("no password data in db for '%s'!" % self._uid)
             return False
-        for i in range(rmin, rmax + 1):
-            if self._data['password'] == hashpwd(pwd, str(i)):
-                # log "AUTHENTICATED after %s!!" % (time() - t1)
+        bcpass = self._data.get('bcpassword', None)
+        if bcpass:
+            if bcpass == bchash(pwd, existing_hash = bcpass):
                 self._authenticated = True
-                break
+        else:
+            for i in range(rmin, rmax + 1):
+                if self._data['password'] == hashpwd(pwd, str(i)):
+                    # log "AUTHENTICATED after %s!!" % (time() - t1)
+                    bcpass = bchash(pwd)
+                    if bcpass:
+                        logger.info("user " + self._uid  +  " logged in with old style password, trying to update")
+                        try:
+                            self._data['bcpassword'] = bcpass
+                            get_users().update_one({'_id': self._uid},
+                                               {'$set':{'bcpassword':bcpass}})
```

Comment:
Shouldn't we also remove the old `password` entry, i.e., the old hash?

Reply:
I considered that, but that would prevent people from logging on using older versions of the LMFDB source that they were running themselves. I have no idea if that would be a problem, but I though it best to not unilaterally make it happen. My plan was to leave the dual system running for a while and then purge the unused ones later.

Reply:
I wouldn't worry very much about that. I have the impression that no one logs in into the LMFDB using their local version. @JohnCremona, what do you think?